### PR TITLE
Cleanup settings._DictProxy and scrapy.telnet

### PR DESCRIPTION
**Pull request by Hanzallah**
This change removes the `_DictProxy` class from `scrapy/settings/__init__.py` since it is no longer used. It also removes `scrapy.telnet` from the `DEPRECATION_RULES` as part of the cleanup. It resolves https://github.com/scrapy/scrapy/issues/5729, resolves https://github.com/scrapy/scrapy/issues/5725.

### CODE REVIEW

File: scrapy/utils/deprecate.py
Comment (Laerte) [inline scrapy/utils/deprecate.py:132]: I think may we should remove this function as well, since now it will just return the provided path, make sense @wRAR?

And probably pass a lambda function to `convert` parameter below (and remove from tests).
https://github.com/scrapy/scrapy/blob/45c2bd7d9c1d6adf0e051842c422c5d6af1d0a17/scrapy/utils/conf.py#L15

Comment (Hanzallah) [reply scrapy/utils/deprecate.py:132]: If there is a possibility of repopulating the `DEPRECATION_RULES` list in the future we should keep the `update_classpath` method otherwise similar logic will need to be re-written.

Comment (Laerte) [reply scrapy/utils/deprecate.py:132]: I think it will be not the case since `@deprecated` decorator already have this functionality:
https://github.com/scrapy/scrapy/blob/45c2bd7d9c1d6adf0e051842c422c5d6af1d0a17/scrapy/utils/decorators.py#L9

But let's see what Scrapy team has to say about.

Comment (wRAR) [reply scrapy/utils/deprecate.py:132]: `update_classpath` is used in the code that parses object dicts such as `DOWNLOADER_MIDDLEWARES`: https://github.com/scrapy/scrapy/blob/45c2bd7d9c1d6adf0e051842c422c5d6af1d0a17/scrapy/utils/conf.py#L15

I think you can't use `@deprecated` for this.

Comment (Laerte) [reply scrapy/utils/deprecate.py:132]: Yeah but the logic depends all on `DEPRECATION_RULES` variable right?

```python
    for prefix, replacement in DEPRECATION_RULES:
        if isinstance(path, str) and path.startswith(prefix):
            new_path = path.replace(prefix, replacement, 1)
            warnings.warn(f"`{path}` class is deprecated, use `{new_path}` instead",
                          ScrapyDeprecationWarning)
            return new_path
    return path
```
If nothing is provided it will just return what was originally send by parameter right?

Comment (wRAR) [reply scrapy/utils/deprecate.py:132]: Right.

**Comment by Hanzallah** [comment]
@wRAR, is this PR ready to be merged?

**Comment by wRAR** [comment]
@Hanzallah we need two approvals from the core contributors to merge a PR.

**Review by wRAR — APPROVED** [review]

**Review by Gallaecio — APPROVED** [review]

File: scrapy/utils/deprecate.py
Comment (Gallaecio) [inline scrapy/utils/deprecate.py:129]: Based on the failing `typing` job, we need to mark this as `List[Tuple[str, str]]` now that the type cannot be inferred from the value.

Comment (Hanzallah) [reply scrapy/utils/deprecate.py]: @Gallaecio I have now added the type annotations to the PR.

**Comment by Gallaecio** [comment]
Great, thanks!

**Pull request merged**
